From eac81abc170deaafaf10e00b47820ed053c733af Mon Sep 17 00:00:00 2001
From: Kurt Newman <kurt.newman@cpanel.net>
Date: Mon, 5 Dec 2016 16:31:36 -0600
Subject: [PATCH] Apply custom cpanel changes

This patch performs the following work:
 - Updates status check to point to whm-server-status
 - Adheres to the cpanel configuration which allows Apache to run
   on a port other than 80
 - Increases ulimit so that servers with large numbers of
   (sub)?domains will operate correctly.
---
 support/apachectl.in | 21 +++++++++++----------
 1 file changed, 11 insertions(+), 10 deletions(-)

index c6ac3ea..f5fb106 100644
--- a/support/apachectl.in
+++ b/support/apachectl.in
@@ -56,7 +56,8 @@ fi
 #
 # the URL to your server's mod_status status page.  If you do not
 # have one, then status and fullstatus will not work.
-STATUSURL="http://localhost:@PORT@/server-status"
+PORT="$(grep 'apache_port=' /var/cpanel/cpanel.config 2>/dev/null | sed -e 's/.*=\([.0-9]*:\)\{0,1\}//;' -e 's/[^0-9]*//g' 2>/dev/null)"
+STATUSURL="http://localhost:${PORT:-80}/whm-server-status"

 # Source /etc/sysconfig/httpd for $HTTPD setting, etc.
 if [ -r /etc/sysconfig/httpd ]; then
@@ -68,7 +69,22 @@ fi
 # number of file descriptors allowed per child process. This is
 # critical for configurations that use many file descriptors,
 # such as mass vhosting, or a multithreaded server.
-ULIMIT_MAX_FILES="@APACHECTL_ULIMIT@"
+
+# Can be overwritten by setting MIN_FD_LIMIT and MAX_FD_LIMIT in /etc/sysconfig/httpd
+if [ -z "$MIN_FD_LIMIT" ]; then
+    MIN_FD_LIMIT=1024
+fi
+
+if [ -z "$MAX_FD_LIMIT" ]; then
+    MAX_FD_LIMIT=16384
+fi
+
+while [ $MIN_FD_LIMIT -lt $MAX_FD_LIMIT ]; do
+    MIN_FD_LIMIT=$(($MIN_FD_LIMIT * 2))
+    ulimit -H -n $MIN_FD_LIMIT 2>/dev/null
+    ulimit -S -n $MIN_FD_LIMIT 2>/dev/null
+done
+
 # --------------------                              --------------------
 # ||||||||||||||||||||   END CONFIGURATION SECTION  ||||||||||||||||||||
 
-- 
2.11.0
